cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java

```diff
@@ -637,14 +594,114 @@ private void populateLabels(Map<String, String> sparkConfMap) {
   }
 
   /**
-   * Deletes Kubernetes namespace and associated resources if they exist for the given CDAP namespace
+   * Checks if namespace already exists from the same CDAP instance. Otherwise, creates a new Kubernetes namespace.
    */
-  private void deleteKubeNamespace(String namespace) throws Exception {
+  private void findOrCreateKubeNamespace(String namespace, String cdapNamespace) throws Exception {
     try {
-      // PropagationPolicy is set to background cascading deletion. Kubernetes deletes the owner object immediately and
-      // the controller cleans up the dependent objects in the background.
-      coreV1Api.deleteNamespace(namespace, null, null, 0, null, "Background", null);
-      LOG.info("Deleted Kubernetes namespace and associated resources for {}", namespace);
+      V1Namespace existingNamespace = coreV1Api.readNamespace(namespace, null, null, null);
+      if (existingNamespace.getMetadata() == null) {
+        throw new IOException(String.format("Kubernetes namespace %s exists but was not created by CDAP", namespace));
+      }
+      Map<String, String> existingNamespaceLabels = existingNamespace.getMetadata().getLabels();
+      if (existingNamespaceLabels == null || !existingNamespaceLabels.containsKey(CDAP_NAMESPACE_LABEL)) {
+        throw new IOException(String.format("Kubernetes namespace %s exists but was not created by CDAP", namespace));
+      } else if (!existingNamespaceLabels.get(CDAP_NAMESPACE_LABEL).equals(cdapNamespace)) {
```

Comment:
You can combine this `else` with the previous if by simply checking 
```
if (existingNamesapceLabels == null || !cdapNamespace.equals(existingNamespaceLabels.get(CDAP_NAMESPACE_LABEL))) {
  throw ...
```

BTW, for local variable, it can be shorter, just as `labels`.